velero-plugin-for-microsoft-azure/common.go

```diff
@@ -29,17 +29,41 @@ const (
 	subscriptionIDEnvVar = "AZURE_SUBSCRIPTION_ID"
 	cloudNameEnvVar      = "AZURE_CLOUD_NAME"
 
-	resourceGroupConfigKey = "resourceGroup"
+	resourceGroupConfigKey   = "resourceGroup"
+	credentialsFileConfigKey = "credentialsFile"
 )
 
-func loadEnv() error {
-	envFile := os.Getenv("AZURE_CREDENTIALS_FILE")
-	if envFile == "" {
+// credentialsFileFromEnv retrieves the Azure credentials file from the environment.
+func credentialsFileFromEnv() string {
+	return os.Getenv("AZURE_CREDENTIALS_FILE")
+}
+
+// selectCredentialsFile selects the Azure credentials file to use, retrieving it
+// from the given config or falling back to retrieving it from the environment.
+func selectCredentialsFile(config map[string]string) (string, error) {
+	if credentialsFile, ok := config[credentialsFileConfigKey]; ok {
+		// Check that the provided credentialsFile exists on disk
+		if _, err := os.Stat(credentialsFile); err != nil {
+			if os.IsNotExist(err) {
+				return "", errors.Wrapf(err, "provided credentialsFile does not exist")
+			}
+			return "", errors.Wrapf(err, "could not get credentialsFile info")
+		}
+		return credentialsFile, nil
+	}
+
+	return credentialsFileFromEnv(), nil
+}
+
+// loadCredentialsIntoEnv loads the variables in the given credentials
+// file into the current environment.
+func loadCredentialsIntoEnv(credentialsFile string) error {
```

Comment:
@carlisia I opted to break this down into a few smaller functions that I can compose together in the object store and volume snapshotter code to hopefully make it a bit clearer which credentials are being used. For example, in the [`Init` function for the volume snapshotter](https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/pull/90/files#diff-4fca62a95ba19c525ebb8734a5057b2d38586ad74a461f60190bb181226b3211R88), we now explicitly state that we are using the credentials from the environment. It's more verbose but to me is more readable. Let me know if you disagree though :)

Reply:
Looks good